internal/channel/plugin.go

```diff
@@ -27,60 +83,96 @@ func (c *Channel) GetInfo() (*plugin.Info, error) {
 	return info, nil
 }
 
-func (c *Channel) SetConfig(config string) error {
-	_, err := c.rpcCall(plugin.MethodSetConfig, json.RawMessage(config))
+// SetConfig sends the setConfig request with given config, returns an error if an error occurred
+func (p *Plugin) SetConfig(config string) error {
+	_, err := p.rpc.Call(plugin.MethodSetConfig, json.RawMessage(config))
 
 	return err
 }
 
-func (c *Channel) SendNotification(contact *recipient.Contact, i contracts.Incident, ev *event.Event, icingaweb2Url string) error {
-	contactStruct := &plugin.Contact{FullName: contact.FullName}
-	for _, addr := range contact.Addresses {
-		contactStruct.Addresses = append(contactStruct.Addresses, &plugin.Address{Type: addr.Type, Address: addr.Address})
+// SendNotification sends the notification, returns an error if fails
+func (p *Plugin) SendNotification(req *plugin.NotificationRequest) error {
+	params, err := json.Marshal(req)
+	if err != nil {
+		return fmt.Errorf("failed to prepare request params: %w", err)
 	}
 
-	if !strings.HasSuffix(icingaweb2Url, "/") {
-		icingaweb2Url += "/"
-	}
+	_, err = p.rpc.Call(plugin.MethodSendNotification, params)
+
+	return err
+}
+
+func terminate(p *Plugin) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	p.logger.Debug("Stopping the channel plugin")
+
+	_ = p.rpc.Close()
+	timer := time.AfterFunc(5*time.Second, func() {
+		p.logger.Debug("killing the channel plugin")
+		_ = p.cmd.Process.Kill()
+	})
+
+	<-p.rpc.Done()
+	timer.Stop()
+	p.logger.Warn("Stopped channel plugin")
+}
 
-	req := plugin.NotificationRequest{
-		Contact: contactStruct,
-		Object: &plugin.Object{
-			Name:      i.ObjectDisplayName(),
-			Url:       ev.URL,
-			Tags:      ev.Tags,
-			ExtraTags: ev.ExtraTags,
-		},
-		Incident: &plugin.Incident{
-			Id:  i.ID(),
-			Url: fmt.Sprintf("%snotifications/incident?id=%d", icingaweb2Url, i.ID()),
-		},
-		Event: &plugin.Event{
-			Time:     ev.Time,
-			Type:     ev.Type,
-			Severity: ev.Severity.String(),
-			Username: ev.Username,
-			Message:  ev.Message,
-		},
+func forwardLogs(errPipe io.Reader, logger *zap.SugaredLogger) {
+	reader := bufio.NewReader(errPipe)
```

Comment:
A `bufio.Scanner` seems to be more fitting here. Its default behavior even covers reading lines.